### PR TITLE
Videopress onboarding v2: domain selection

### PR DESCRIPTION
**Pull request by thedebian**
#### Proposed Changes

This PR adds the domain selection step in the videopress flow.

#### Testing Instructions
* Apply this PR
* Go to http://calypso.localhost:3000/setup/intro?flow=videopress
* ✅ If you're logged in, you should be redirected to the options step
* ✅ If you're not logged in, you should be redirected to the account creation step, then to the options step
* Add a site title and click continue
* ✅ You should be redirected to the domain selection step, with the site title as suggestion

Fixes 1293-gh-Automattic/greenhouse

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/choose-a-domain/index.tsx [hunk 1]
```diff
@@ -1,21 +1,87 @@
...
+			return undefined;
+		}
+
+		return (
+			<FormattedHeader
+				id={ 'choose-a-domain-header' }
+				headerText="Choose a domain"
+				subHeaderText={
+					<>
+						Make your video site shine with a custom domain. Not sure yet ?{ ' ' }
```
Comment (roundhill) [inline client/landing/stepper/declarative-flow/internals/steps-repository/choose-a-domain/index.tsx:57]: Localize these strings?

**Review by roundhill — CHANGES_REQUESTED** [review]
Worked well but can the display strings be localized?

File: client/landing/stepper/declarative-flow/internals/steps-repository/choose-a-domain/index.tsx (hunk 1 above)
Comment (thedebian) [reply client/landing/stepper/declarative-flow/internals/steps-repository/choose-a-domain/index.tsx:57]: I added them in #67596 👍

**Review by pgk — APPROVED** [review]
worked as expected!

**Pull request merged**